Terminal.Gui/Text/TextFormatter.cs

```diff
@@ -1683,13 +1695,13 @@ private static List<Rune> PerformCorrectFormatDirection (TextDirection textDirec
     private static List<string> PerformCorrectFormatDirection (TextDirection textDirection, List<string> lines)
     {
         return textDirection switch
-               {
-                   TextDirection.TopBottom_RightLeft
-                       or TextDirection.LeftRight_BottomTop
-                       or TextDirection.RightLeft_BottomTop
-                       or TextDirection.BottomTop_RightLeft => lines.ToArray ().Reverse ().ToList (),
-                   _ => lines
-               };
+        {
+            TextDirection.TopBottom_RightLeft
+                or TextDirection.LeftRight_BottomTop
+                or TextDirection.RightLeft_BottomTop
+                or TextDirection.BottomTop_RightLeft => lines.ToArray ().Reverse ().ToList (),
```

Comment:
This is expensive, because there are two explicit materializations that force enumeration and copies, from the ToArray() and the ToList(), on top of the work Reverse already has to do.

At least drop the first ToArray(), since `List<T>` already has `Reverse()`.

It would also be better to drop the ToList() as well, and return IEnumerable, so that enumeration/evaluation of the result doesn't happen until the last second where used.

Reply:
Not relevant for this PR. Not touching this in this PR.